src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -795,14 +795,18 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
     
     public List<String> getArchivedCourseIds(List<CourseDetailsBundle> allCourses, List<InstructorAttributes> instructorList) {
         List<String> archivedCourseIds = new ArrayList<String>();
-        for (CourseDetailsBundle cdb : allCourses) {
-            if (cdb.course.isArchived) {
-                archivedCourseIds.add(cdb.course.id);
-            }
-        }
         for (InstructorAttributes instructor : instructorList) {
             if (instructor.isArchived != null && instructor.isArchived == true) {
                 archivedCourseIds.add(instructor.courseId);
+            } else if (instructor.isArchived == null) {
+                for (CourseDetailsBundle cdb : allCourses) {
+                    if (cdb.course.id == instructor.courseId) {
+                        if (cdb.course.isArchived) {
+                            archivedCourseIds.add(cdb.course.id);
+                        }
```

Comment:
This is hard to read. We have a <code>isCourseArchived</code> in Logic, we can move that method into CoursesLogic, and use that here?